core/unit_test/TestSharedHostPinnedSpace.cpp

```diff
@@ -0,0 +1,132 @@
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+
+namespace {
+
+template <typename ViewType>
+struct Increment {
+  ViewType view_;
+
+  template <typename ExecutionSpace>
+  explicit Increment(ExecutionSpace, ViewType view) : view_(view) {
+    Kokkos::fence();
+    Kokkos::parallel_for(
+        "increment",
+        Kokkos::RangePolicy<ExecutionSpace, Kokkos::IndexType<size_t>>{
+            0, view_.size()},
+        *this);
+    Kokkos::fence();
+  }
+
+  KOKKOS_FUNCTION
+  void operator()(const size_t idx) const { ++view_(idx); }
+};
+
+template <typename ViewType>
+struct CheckResult {
+  ViewType view_;
+  const int targetVal_;
+  unsigned error = 0;
+
+  template <typename ExecutionSpace>
+  CheckResult(ExecutionSpace, ViewType view, int targetVal)
+      : view_(view), targetVal_(targetVal) {
+    Kokkos::parallel_reduce(
+        "check",
+        Kokkos::RangePolicy<ExecutionSpace, Kokkos::IndexType<size_t>>{
+            0, view_.size()},
+        *this, Kokkos::Sum<unsigned>(error));
```

Comment:
Does CTAD not kick in?
```suggestion
        *this, Kokkos::Sum(error));
```

Reply:
```
/kokkos/core/unit_test/TestSharedHostPinnedSpace.cpp:81:16: error: no viable constructor or deduction guide for deduction of template arguments of 'Sum'
        *this, Kokkos::Sum(numErrors));
```

Reply:
All built-in reducers have also a memory space template argument.